cdap-ui/app/cdap/services/datasource/index.js

```diff
@@ -35,17 +35,20 @@ import 'rxjs/add/operator/debounceTime';
 import WindowManager, { WINDOW_ON_BLUR, WINDOW_ON_FOCUS } from 'services/WindowManager';
 import { objectQuery } from 'services/helpers';
 import ifvisible from 'ifvisible.js';
+import SystemDelayStore from 'services/SystemDelayStore';
+import SystemDelayActions from 'services/SystemDelayStore/SystemDelayActions';
+import cloneDeep from 'lodash/cloneDeep';
 
 const CDAP_API_VERSION = 'v3';
 // FIXME (CDAP-14836): Right now this is scattered across node and client. Need to consolidate this.
 const REQUEST_ORIGIN_ROUTER = 'ROUTER';
 
 export default class Datasource {
-  constructor(genericResponseHandlers = [() => true]) {
-    this.eventEmitter = ee(ee);
+  constructor(genericResponseHandlers = [() => true],options) {
+  this.eventEmitter = ee(ee);
     let socketData = Socket.getObservable();
     this.bindings = {};
-
+    this.excludeFromHealthCheck = options ? !!options.excludeFromHealthCheck : false;
```

Comment:
this seems a little weird. Why is the toggle for excluding from health check happening for a datasource instance? We want to be able to exclude a single API call from the healthCheck, so this toggle needs to come from individual bindings, not for the entire instance of datasource

Reply:
fixed to specify this option at API endpoint level instead of at datasource level